player/src/lib.rs

```diff
@@ -209,8 +209,8 @@ impl GlobalPlay for wgc::hub::Global<IdentityPassThroughFactory> {
                 self.device_create_bind_group_layout::<B>(
```

Comment:
The main motivator behind `Cow` was actually to let the tracing use the same types. Currently `src/device/trace.rs` have to redefine a bunch of types to be owned. If we switch `wgt` stuff to `Cow`, we can use it directly in the `trace` module.
For example this variant we are processing here could just be `CreatebindGroupLayout(Id<id::BindGroupLayoutId>, wgt::BindGroupLayoutDescriptor<'a>)`. The current approach with named fields was different because we couldn't reuse the descriptor types. Once we can, there is no need to name fields have anything other than `(id, desc)` tuple variants.

Reply:
Ah, I wasn't aware of that.

Reply:
I've now mostly reduced the duplication among the tracing API types. The player can now simply pass the descriptors from `Action` into the right wgpu APIs, and owned variants with `'static` params can be created via the new `ToStatic` trait.

Sadly the implementation of `ToStatic` is relativly verbose, as its a recursivly delegating trait like `Clone` - but that also means that we could turn it into an relativly simple custom derive attribute that lives on crates.io if we wanted to remove that boilerplate.